src/models/blueprint-cloner.js

```diff
@@ -0,0 +1,84 @@
+import path from 'path';
+import { copySync } from 'fs-extra';
+import walkSync from 'walk-sync';
+
+import { fileExists } from '../util/fs';
+
+const FILES_BLACKLIST = ['.ds_store', '.git', '.gitkeep'];
+
+export default class BlueprintCloner {
+  constructor(blueprint, options) {
+    this.blueprint = blueprint;
+    this.options = options || {};
+    this.ui = options.ui;
+  }
+
+  clone() {
+    const blueprint  = this.blueprint;
```

Comment:
By the time a blueprint is added to the collection, it will already have a validated existent path to itself.  And will also have a method that returns it's files.

Reply:
Awesome!

I'll make the changes once the new api is ready.